### ISSUE TITLE
Resolution of relative path Avro schema files doesn't work properly

### ISSUE THREAD

**Issue by dimarzio** [Pakisan/jasyncapi-idea-plugin#37]
Hello Pavel,

I have a strange issue with local path resolution of locally stored *.avsc files. And actually only with these files.

In the AsyncAPI definition file (*.yml) I have multiple interface channels where I'm using local files references for examples and schema payloads. The definition looks like this:

asyncapi: 2.3.0
channels:
  kafka-topic-name:
    publish:
      message:
        name: message_name
        description: interface description
        schemaFormat: 'application/vnd.apache.avro;version=1.9.0'
        examples:
          - name: message_example_name
            summary: message example summary
            payload:
              $ref: "./example-folder/filename.json"
        payload:
          $ref: './schema-folder/LocationUpdated.avsc'

When I'm starting browser preview I see following error message: "Error: Error downloading http://localhost:63342/asyncapi/resources?referenceUrl=/absolute/path/to/the/avro/schema/file/filename.avsc HTTP ERROR 404"

When I'm putting the Avro schema definition directly into the AsyncAPI definition file it works again. The resolution of the example JSON file works in every case.

So I'm not 100-% sure if it's a plugin or AsyncAPI avro parser bug.

When I'm for example running the asyncapi generator (html template) it works without any problem and the external *.avsc files will be properly resolved from the relative path and also properly parsed.

Could you please take a look at that?

Thank you,
Dmitriy

**Comment by Pakisan** [comment]
@dimarzio hi.

I will check it. But I think that is not a bug, but limitation because, of allowance only of `.yml`, `.json` files. Looks I don't thought about Avro.

Anyway will implement Avro in `1.6.0`

**Comment by dimarzio** [comment]
@Pakisan Hi,

Thanks a lot for the response! Than consider this issue as a feature request :)

Thank you,
Dmitriy

**Comment by Pakisan** [comment]
@dimarzio

```json
{
  "asyncapi": "2.5.0",
  "info": {
    "title": "Account Service",
    "version": "1.0.0",
    "description": "This service is in charge of processing user signups :rocket:"
  },
  "channels": {
    "user/signedup": {
      "subscribe": {
        "message": {
          "$ref": "#/components/messages/UserSignedUp"
        }
      },
      "publish": {
        "message": {
          "$ref": "#/components/messages/UserSignedUp"
        }
      }
    },
    "user/signedout": {
      "subscribe": {
        "message": {
          "$ref": "#/components/messages/UserSignedUp"
        }
      },
      "publish": {
        "message": {
          "$ref": "#/components/messages/UserSignedUp"
        }
      }
    }
  },
  "components": {
    "messages": {
      "UserSignedOut": {
        "schemaFormat": "application/vnd.apache.avro;version=1.9.0",
        "payload": {
          "$ref": "./avro/userSignedOutPayload.avsc"
        }
      },
      "UserSignedUp": {
        "schemaFormat": "application/vnd.apache.avro;version=1.9.0",
        "payload": {
          "$ref": "./avro/userSignedUpPayload.avsc"
        }
      }
    }
  }
}
```

```json
{
  "type": "record",
  "namespace": "accountService",
  "name": "UserSignedUpPayload",
  "fields": [
    {
      "name": "displayName",
      "type": "String",
      "doc": "Name of the user"
    }
  ]
}
```

```json
{
  "type": "record",
  "namespace": "accountService",
  "name": "UserSignedUpPayload",
  "fields": [
    {
      "name": "displayName",
      "type": "String",
      "doc": "Name of the user"
    },
    {
      "name": "email",
      "type": "string",
      "doc": "Email of the user"
    }
  ]
}
```

Implemented in 1.7.0
